### DISCUSSION TITLE
Not Even Rect and Circle Border.

### DISCUSSION THREAD

**Discussion by RathodAnnu13** [fabricjs/fabric.js#10064] [Q&A]
I am currently using Fabric.js version 5.3.0. When I attempt to load a Rect or Circle with a border and an opacity of 0.7 using the loadFromJSON() method, 
I encounter the following issue with the output canvas.

![image](https://github.com/user-attachments/assets/48b54229-b9ea-4e07-bdb1-6c0606a04b15)

Please find the JSON code below.
` "canvasJSON": {
        "version": "5.3.0",
        "objects": [
          {
            "type": "rect",
            "version": "5.3.0",
            "originX": "left",
            "originY": "top",
            "left": 51,
            "top": 26,
            "width": 357,
            "height": 357,
            "fill": "#0F1934",
            "stroke": "#FFDE59",
            "strokeWidth": 50,
            "strokeDashArray": null,
            "strokeLineCap": "butt",
            "strokeDashOffset": 0,
            "strokeLineJoin": "miter",
            "strokeUniform": false,
            "strokeMiterLimit": 4,
            "scaleX": 1,
            "scaleY": 1,
            "angle": 0,
            "flipX": false,
            "flipY": false,
            "opacity": 0.7,
            "shadow": {
              "color": "#000000ff",
              "blur": 0,
              "offsetX": 0,
              "offsetY": 0,
              "affectStroke": false,
              "nonScaling": false
            },
            "visible": true,
            "backgroundColor": "",
            "fillRule": "nonzero",
            "paintFirst": "fill",
            "globalCompositeOperation": "source-over",
            "skewX": 0,
            "skewY": 0,
            "rx": 67,
            "ry": 67,
            "id": 847917,
            "excludeFromExport": false,
            "_controlsVisibility": {
              "tl": true,
              "tr": true,
              "br": true,
              "bl": true,
              "ml": true,
              "mt": true,
              "mr": true,
              "mb": true,
              "mtr": true
            },
            "isLocked": false,
            "element_type": "basicShape",
            "index": 1,
            "isObjectcenter": false
          },
          {
            "type": "circle",
            "version": "5.3.0",
            "originX": "left",
            "originY": "top",
            "left": 266,
            "top": 439,
            "width": 390,
            "height": 390,
            "fill": "#0F1934",
            "stroke": "#FFDE59",
            "strokeWidth": 32,
            "strokeLineCap": "butt",
            "strokeDashOffset": 0,
            "strokeLineJoin": "miter",
            "strokeUniform": false,
            "strokeMiterLimit": 4,
            "scaleX": 1,
            "scaleY": 1,
            "angle": 0,
            "flipX": false,
            "flipY": false,
            "opacity": 0.7,
            "shadow": {
              "color": "#000000ff",
              "blur": 0,
              "offsetX": 0,
              "offsetY": 0,
              "affectStroke": false,
              "nonScaling": false
            },
            "visible": true,
            "backgroundColor": "",
            "fillRule": "nonzero",
            "paintFirst": "fill",
            "globalCompositeOperation": "source-over",
            "skewX": 0,
            "skewY": 0,
            "radius": 159.055292253,
            "startAngle": 0,
            "endAngle": 360,
            "id": 469484,
            "excludeFromExport": false,
            "_controlsVisibility": {
              "tl": true,
              "tr": true,
              "br": true,
              "bl": true,
              "ml": true,
              "mt": true,
              "mr": true,
              "mb": true,
              "mtr": true
            },
            "isLocked": false,
            "element_type": "basicShape",
            "isObjectcenter": false,
            "index": 0
          }
        ]
      }`

**Comment by asturur** [comment]
And the issue is?

**Reply by RathodAnnu13** [reply]
The issue is that the rect's border should display a uniform color throughout.
As i shown below.

![image](https://github.com/user-attachments/assets/bd3b5fdc-331b-4482-894d-8df1b9282b9c)

**Reply by asturur** [reply]
That is not how canvas work in general. You can enable caching to force that maybe, but it won't work if you use an opacity for the stroke color itself.
